### ISSUE TITLE
Ai Chat (Google Auth Provider): userId redis issues

### ISSUE THREAD

**Issue by borjadotai** [vercel-labs/ai-chatbot#176]
Hey I was trying out your ai chat template and it is pretty amazing, however, when I changed the auth provider to instead of Github, some things started breaking for the weirdest reasons, I have found a workaround that fixes it but thought I'd let you guys know about it. The root cause was KV / Upstash / Redis, I wrote an issue explaining everything on their repo.

[https://github.com/upstash/upstash-redis/issues/699](https://github.com/upstash/upstash-redis/issues/699)

TLDR: hgetall turns the userId string (which for google is much longer) into a number and rounds it, causing bugs.
Temporary solution was to use both `hgetall` and `hget`

```
let rawChat = await kv.hgetall<Chat>(`chat:${id}`)
const uid = await kv.hget(`chat:${id}`, 'userId')
const chat = { ...rawChat!, userId: uid }
```
